src/gar.lua

```diff
@@ -1836,23 +1836,22 @@ function gar.redelegateStake(params)
 	-- Move redelegation fee to protocol balance
 	balances.increaseBalance(ao.id, redelegationFee)
 
-	Redelegations[delegateAddress] = {
-		timestamp = currentTimestamp,
-		redelegations = (previousRedelegations and previousRedelegations.redelegations or 0) + 1,
-	}
+	local redelegationsSinceFeeReset = (previousRedelegations and previousRedelegations.redelegations or 0) + 1
 
-	-- update the gateway
+	-- update the source and target gateways, and the delegator's redelegation fee data
 	GatewayRegistry[sourceAddress] = sourceGateway
-
-	-- update the target gateway
 	GatewayRegistry[targetAddress] = targetGateway
+	Redelegations[delegateAddress] = {
+		timestamp = currentTimestamp,
+		redelegations = redelegationsSinceFeeReset,
+	}
 
 	return {
-		sourceGateway = sourceGateway,
-		targetGateway = targetGateway,
+		sourceAddress = sourceAddress,
+		targetAddress = targetAddress,
```

Comment:
i was suggesting, just changing this in the Handler result 😄 , but this is fine too